runtime/src/bank_forks.rs

```diff
@@ -378,4 +390,74 @@ mod tests {
         bank_forks.insert(child_bank);
         assert_eq!(bank_forks.active_banks(), vec![1]);
     }
+
+    #[test]
+    fn test_bank_forks_different_set_root() {
+        solana_logger::setup();
+        let leader_keypair = Keypair::new();
+        let GenesisConfigInfo {
+            mut genesis_config,
+            mint_keypair: _,
+            voting_keypair,
+        } = create_genesis_config_with_leader(10_000, &leader_keypair.pubkey(), 1_000);
+        let slots_in_epoch = 32;
+        genesis_config.epoch_schedule = EpochSchedule::new(slots_in_epoch);
+
+        let bank0 = Bank::new(&genesis_config);
+        let mut bank_forks0 = BankForks::new(bank0);
+
+        let bank1 = Bank::new(&genesis_config);
+        let mut bank_forks1 = BankForks::new(bank1);
+        bank_forks0.set_root(0, &None, None);
```

Comment:
nit: move `set_root` after `let mut bank_forks0 = BankForks::new(bank0);`  be closer to the declaration site

Reply:
Done https://github.com/solana-labs/solana/pull/13477/commits/1721cb69af4c9f50fa98d638a7a9a15883914395